src/simplewallet/simplewallet.cpp

```diff
@@ -6379,7 +6388,27 @@ static bool parse_lns_name_string(std::vector<std::string> const &args, size_t f
   return true;
 }
 //----------------------------------------------------------------------------------------------------
-bool simple_wallet::buy_lns_mapping(const std::vector<std::string>& args)
+std::string eat_named_argument(std::vector<std::string> &args, char const *prefix, size_t prefix_len)
+{
+  std::string result = {};
+  for (auto it = args.begin(); it != args.end(); it++)
+  {
+    if (it->size() > prefix_len && memcmp(it->data(), prefix, prefix_len) == 0)
```

Comment:
This is basically a repeat of my comment from the other PR: take the `prefix` as a `string_view` and then:
```C++
if (string_view{*it}.substr(0, prefix.size()) == prefix) {
  result = it->substr(prefix.size());
  ...
```
(No need to specify a length for a to-the-end substr, and no need to worry about "overshooting" the length parameter: both are well defined to return up-to-the-end for both `std::string` and `string_view`).